Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabDomUtils.cpp

```diff
@@ -582,6 +582,113 @@ namespace AzToolsFramework
                         "PrefabDomUtils::AddNestedInstance - Nested instance already exists in prefab DOM.");
                 }
             }
+
+            bool SubstituteInvalidParentsInEntities(PrefabDom& templateDomRef)
+            {
+                // Search for a TransformComponent and check the "Parent Entity" statement
+                auto sourceIt = templateDomRef.FindMember(PrefabDomUtils::SourceName);
+                if (sourceIt == templateDomRef.MemberEnd() || !sourceIt->value.IsString())
+                {
+                    AZ_Warning("Prefab", false, "lacks a 'Source' container.");
+                    return false;
+                }
+                const AZStd::string path(sourceIt->value.GetString());
+
+                const auto containerEntityIt = templateDomRef.FindMember(PrefabDomUtils::ContainerEntityName);
+                if (containerEntityIt == templateDomRef.MemberEnd() || !containerEntityIt->value.IsObject())
+                {
+                    AZ_Warning("Prefab", false, "'%s' lacks a 'ContainerEntity' container.", path.c_str());
```

Comment:
Some of these warnings are inconsistent with other errors here.

In general, I prefer
* Trace or Printf if the problem found was automatically fixed and will cause no issues and user doesn't need to know
* Warning if the problem found was automatically fixed or could not be fixed, but the asset will continue to function as expected, albeit with potential performance issues or user action required.
* Error if the operation failed, and the asset in question is no longer usable and will suffer functional problems, like, for example, failing to load the collision of a mesh which will result in a player falling thru the world, or cause crashes later
* Assert if the problem exists in code and can only be fixed by changing code.

Unit tests that expect errros and warnings have macros to ignore or require them.

In this case, the prefab seems completely broken and its early out as "return false", should it not be an error?

Reply:
> Unit tests that expect errros and warnings have macros to ignore or require them.
>
> In this case, the prefab seems completely broken and its early out as "return false", should it not be an an error?

If the UnitTests had comments on which expected errors and warnings were already beeing suppressed before this change, I'd definitely go this way.